File: 217-ContainsDuplicate/217-ContainsDuplicate.py
# Last updated: 7/18/2025, 7:00:42 PM
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """ 
        seen = {}
        for i, num in enumerate(nums):
            if num in seen:
                return True
            seen[num] = i
        
        return False
            

        

File: 217-ContainsDuplicate/217-ContainsDuplicate.java
// Last updated: 7/26/2025, 11:36:43 PM
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        for (int num : nums){
            if( seen.contains(num) ) return true;
            seen.add(num);
        }
        return false;
    }
}